chromadb/telemetry/events.py

```diff
@@ -13,11 +13,76 @@ class ServerStartEvent(TelemetryEvent):
     name: ClassVar[str] = "server_start"
 
 
+@dataclass
+class ClientCreateCollectionEvent(TelemetryEvent):
+    name: ClassVar[str] = "client_create_collection"
+    collection_uuid: str
+    embedding_function: str
+
+
 @dataclass
 class CollectionAddEvent(TelemetryEvent):
     name: ClassVar[str] = "collection_add"
     collection_uuid: str
     add_amount: int
+    with_embeddings: bool
+    with_metadata: bool
+    with_documents: bool
+    batch_size = 4
+
+    def can_batch(self, other: TelemetryEvent) -> bool:
+        return (
+            isinstance(other, CollectionAddEvent)
+            and self.collection_uuid == other.collection_uuid
+            and self.with_embeddings == other.with_embeddings
+            and self.with_metadata == other.with_metadata
+            and self.with_documents == other.with_documents
+        )
+
+    def batch(self, other: "TelemetryEvent") -> "CollectionAddEvent":
+        if not self.can_batch(other):
+            raise ValueError("Cannot batch events")
+        other = cast(CollectionAddEvent, other)
+        return CollectionAddEvent(
+            collection_uuid=self.collection_uuid,
+            add_amount=self.add_amount + other.add_amount,
+            with_embeddings=self.with_embeddings,
+            with_metadata=self.with_metadata,
+            with_documents=self.with_documents,
+        )
+
+
+@dataclass
+class CollectionUpdateEvent(TelemetryEvent):
+    name: ClassVar[str] = "collection_update"
```

Comment:
could we do `self.__class__.__name__` on `TelemetryEvent` ? Might save book keeping.

Reply:
We can't -- because it's a class variable we don't have a `self` to call that on in `TelemetryEvent`. I tried setting it in `__init__` but the type checker yelled at me for setting a class variable via instance

Reply:
sorry, i mean just do `__class__.__name__` on the instance, not on `self`